mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -231,6 +240,85 @@ def _azure_upload_file(self, credentials, local_file, artifact_path):
         except Exception as err:
             raise MlflowException(err)
 
+    def _azure_adls_gen2_upload_file(self, credentials, local_file, artifact_path):
+        """
+        Uploads a file to a given Azure storage location using the ADLS gen2 API.
+        """
+        try:
+            headers = self._extract_headers_from_credentials(credentials.headers)
+
+            # try to create the file
+            try:
+                put_adls_file_creation(credentials.signed_uri, headers=headers)
+            except requests.HTTPError as e:
+                if e.response.status_code in [401, 403]:
```

Comment:
Please make sure these are the only codes will be returned on authorization failure scenarios.

Reply:
It actually looks like there are no 401's returned?  https://learn.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create#response

Reading through these codes, the one that would trigger on a potential token expiration is one of the 403's.  I'll change to limit to just 403.